lib/conjur/command/plugin.rb

```diff
@@ -0,0 +1,135 @@
+# the Software, and to permit persons to whom the Software is furnished to do so,
+# subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+# FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+require 'conjur/command'
+require 'yaml'
+
+GEMBIN = '/opt/conjur/embedded/bin/gem'
+
+class Conjur::Command::Plugin < Conjur::Command
+  desc 'Manage plugins'
+  command :plugin do |cmd|
+    # Only enable this plugin for omnibus-installed CLI
+    hide_docs(cmd) unless File.file?(GEMBIN) or ENV['DEV']
+
+    cmd.desc 'List installed plugins'
+    cmd.command :list do |c|
+      c.action do
+        Conjur::Config.load
+        Conjur::Config.plugins.each do |p|
+          begin
+            gem = Gem::Specification.find_by_name "conjur-asset-#{p}"
+            puts "#{p} (#{gem.version})"
+          rescue Gem::LoadError
+            nil
+          end
+        end
+      end
+    end
+
+    cmd.desc 'Install a plugin'
+    cmd.arg_name 'name'
+    cmd.command :install do |c|
+      c.arg_name 'version'
+      c.desc 'Version of the plugin to install'
+      c.flag [:v, :version], :default_value => 'latest'
+
+      c.action do |_, options, args|
+        install_plugin(require_arg(args, 'name'), options[:version])
+      end
+    end
+
+    cmd.desc 'Uninstall a plugin'
+    cmd.arg_name 'name'
+    cmd.command :uninstall do |c|
+      c.action do |_, _, args|
+        name = require_arg(args, 'name')
+        uninstall_plugin(name)
+      end
+    end
+
+    cmd.desc "Show a plugin's details"
+    cmd.arg_name 'name'
+    cmd.command :show do |c|
+      c.action do |_, _, args|
+        name = require_arg(args, 'name')
+        begin
+          gem = Gem::Specification.find_by_name "conjur-asset-#{name}"
+          puts "Name: #{name}"
+          puts "Description: #{gem.summary}"
+          puts "Gem: #{gem.name}"
+          puts "Version: #{gem.version}"
+        rescue Gem::LoadError
+          puts "Plugin '#{name}' is not installed"
+        end
+      end
+    end
+  end
+end
+
+def install_plugin(name, version)
+  gem_name = name.start_with?('conjur-asset-') ? name : "conjur-asset-#{name}"
+
+  gem = Gem.latest_spec_for(gem_name)
+  if gem.nil?
+    exit_now! "gem #{gem_name} does not exist"
+  end
+
+  # Uninstall all versions of plugin to avoid multiple versions
+  uninstall_plugin(name)
+
+  version = gem.version if version == 'latest'
+  okay = system("#{GEMBIN} install #{gem.name} -v #{version}")
```

Comment:
Check out using Gem::DependencyInstaller instead of `system`

https://github.com/rdp/specific_install/blob/master/lib/rubygems/commands/specific_install_command.rb#L152

http://ruby-doc.org/stdlib-1.9.3/libdoc/rubygems/rdoc/Gem/DependencyInstaller.html